blazar/api/v1/networks/service.py

```diff
@@ -66,3 +67,29 @@ def delete_network(self, network_id):
         :type network_id: str
         """
         self.manager_rpcapi.delete_network(network_id)
+
+    @policy.authorize('networks', 'list_allocations')
+    def list_allocations(self, query):
+        """List all allocations on all network segments.
+
+        :param query: parameter to query allocations
+        :type query: dict
+        """
+        ctx = context.current()
+        detail = False
+
+        if policy.enforce(ctx, 'admin', {}, do_raise=False):
+            detail = True
+
+        return self.manager_rpcapi.list_allocations(query, detail=detail)
+
+    @policy.authorize('networks')
+    def get_allocations(self, network_id, query):
+        """List all allocations on a specificied network segment.
+
+        :param network_id: ID of the network segment in Blazar BDself.
+        :type network_id: str
+        :param query: parameters to query allocation
+        :type query: dict
+        """
+        return self.manager_rpcapi.get_allocations(network_id, query)
```

Comment:
Suggestion: make RPC only support list_allocations and just call it here and pass network_id as an argument. Then return the first element of the list (should only be 1 entry in array).